samples/query-features-with-arcade-expression/src/main/java/com/esri/arcgismaps/sample/queryfeatureswitharcadeexpression/components/QueryFeaturesWithArcadeExpressionViewModel.kt

```diff
@@ -0,0 +1,190 @@
+        // update the marker location to where the user tapped on the map
+        markerGraphic.geometry = point
+        viewModelScope.launch {
+            // centre the viewpoint on where the user tapped on the map
+            mapViewProxy.setViewpointCenter(point)
+
+            // evaluate an Arcade expression on the tapped screen coordinate
+            evaluateArcadeExpression(screenCoordinate)
+        }
+    }
+
+    /**
+     * Evaluates an Arcade expression that returns crime in the last 60 days at the tapped
+     * [screenCoordinate] on the [arcGISMap] with the [policeBeatsLayer] and outputs the result
+     * to the [queryStateFlow] property.
+     */
+    private suspend fun evaluateArcadeExpression(screenCoordinate: ScreenCoordinate) {
+        policeBeatsLayer?.let { layer ->
+            // show the loading spinner as the Arcade evaluation can take time to complete
+            _queryStateFlow.value = QueryState(loadState = LoadState.LOADING)
+
+            // identify the layer and its elements based on the position tapped on the mapView and
+            // get the result
+            val result = mapViewProxy.identify(
+                layer = layer,
+                screenCoordinate = screenCoordinate,
+                tolerance = 12.dp,
+                returnPopupsOnly = false
+            )
+
+            // get the result as an IdentifyLayerResult
+            val identifyLayerResult = result.getOrElse { error ->
+                // if the identifyLayer operation failed show an error and return
+                messageDialogVM.showMessageDialog(
+                    "Error identifying layer:",
+                    error.message.toString()
+                )
+                // reset the query results and loading indicator
+                _queryStateFlow.value = QueryState()
+                return
+            }
+
+            if (identifyLayerResult.geoElements.isEmpty()) {
+                _queryStateFlow.value = QueryState(loadState = LoadState.LOADED)
+                return
+            }
+
+            // get the first identified GeoElement as an ArcGISFeature
+            val identifiedFeature = identifyLayerResult.geoElements.first() as ArcGISFeature
+            // create a string containing the Arcade expression
+            val expressionValue = "var crimes = FeatureSetByName(\$map, 'Crime in the last 60 days');\n" +
+                    "return Count(Intersects(\$feature, crimes));"
+
+            // create an arcade expression from the string and configure an arcade evaluator
+            val arcadeExpression = ArcadeExpression(expressionValue)
+            val arcadeEvaluator = ArcadeEvaluator(arcadeExpression, ArcadeProfile.FormCalculation)
+
+            // map profile variables with the feature
+            val profileVariables = mapOf<String, Any>("\$feature" to identifiedFeature, "\$map" to arcGISMap)
+            val evaluationResult = arcadeEvaluator.evaluate(profileVariables)
```

Comment:
As someone who's not familiar with using Arcade expressions, I think the commenting here could be more helpful. How about something like this?
```
            // create a map of profile variables with the feature and arcGISMap as key value pairs
            val profileVariables = mapOf<String, Any>("\$feature" to identifiedFeature, "\$map" to arcGISMap)

            // evaluate using these profile variables and get the result
            val evaluationResult = arcadeEvaluator.evaluate(profileVariables)
```